api/v1alpha1/account_types.go

```diff
@@ -48,6 +48,9 @@ type AccountSpec struct {
 	Creator *string `json:"creator,omitempty"`
 
 	Extensions []Extension `json:"extensions,omitempty"`
+
+	// Additional account metadata information that should be stored with the account
+	AccountMetadata *apiextensionsv1.JSON `json:"accountMetadata,omitempty"`
```

Comment:
Maybe to avoid using the term metadata that is already in use, lets just use a generic `data` property